mne/epochs.py

```diff
@@ -127,26 +127,20 @@ def _save_split(epochs, fname, part_idx, n_parts, fmt, split_naming, overwrite):
     """
     # insert index in filename
     base, ext = op.splitext(fname)
-    if part_idx > 0:
-        if split_naming == "neuromag":
-            fname = "%s-%d%s" % (base, part_idx, ext)
-        else:
-            assert split_naming == "bids"
-            fname = _construct_bids_filename(base, ext, part_idx, validate=False)
-            _check_fname(fname, overwrite=overwrite)
+    if part_idx > 0 and split_naming == "neuromag":
+        fname = f"{base}-{part_idx:d}{ext}"
+    elif split_naming == "bids" and n_parts > 1:
+        fname = _construct_bids_filename(base, ext, part_idx + 1)
+    _check_fname(fname, overwrite=overwrite)
```

Comment:
call _check_fname for both 'neuromag' and 'bids' to fix overwriting bug when 
`epochs.save('test-epo.fif', overwrite=False)` would overwrite existing 'test-epo-1.fif'